### PR TITLE
Switch to modern way of Python packaging

### LINKED ISSUE #27: Switch to modern way of Python packaging
See: https://packaging.python.org/en/latest/tutorials/packaging-projects/

- [x] I volunteer to implement it.

### PR DESCRIPTION
**Pull request by rominf**
Fix #27.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,36 +1,9 @@
...
-        "Intended Audience :: System Administrators",
-        "Programming Language :: Python :: 3.7",
-        "Natural Language :: English",
-    ],
-)
+# a setup.py shim which is required for an editable install
+# see pyproject.toml for build configuration
+# https://peps.python.org/pep-0518
+import setuptools
+￼
```
Comment (rominf) [inline setup.py]: I'll remove this symbol in next push.

File: Makefile [hunk 2]
```diff
@@ -12,24 +9,6 @@ clean:
 	$(RM) -r *.egg-info/ build/ dist/ rpm/
 	$(RM) ../rpm_s3_mirror_* test-*.xml
 
-rpm:
```
Comment (lionbee) [inline Makefile:15]: Was it on purpose that we do not have a mke target to build an rpm anymore?

**Review by lionbee — COMMENTED** [review]
Overall it looks good

File: Makefile (hunk 2 above)
Comment (rominf) [reply Makefile:15]: Yes. RPM building will be done downstream. Also, RPM building was not done properly anyway, proper way of RPM building is to do it per OS distribution.

**Review by facetoe — APPROVED** [review]

**Pull request merged**